### DISCUSSION TITLE
Can't subscribe to feed that works in cURL

### DISCUSSION THREAD

**Discussion by danburd** [FreshRSS/FreshRSS#6520] [Q&A] [answered]
Hello, since May 20 I can't subscribe to a feed that appears to work in cURL. Note that the site seems to expect a particular set of user-agents.

This command properly captures the feed content:
```curl https://www.thenewhumanitarian.org/rss/all.xml -A "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"```

But subscribing to the feed in FreshRSS, with the user-agent field properly customized, gives me the error: ```Blast! This feed has encountered a problem. Please verify that it is always reachable then update it.```

Any help would be appreciated.

**Comment by Alkarex** [comment, marked as answer]
Hello,
It is again a feed broken due to Cloudflare (we have several other issues on the topic):

```
< cf-mitigated: challenge
< server: cloudflare
```

The filter is based on a combination of several parameters.

For instance, the following works for me, but not with normal curl (your command is not working for me):

```sh
docker run --rm lwthiker/curl-impersonate:0.6-chrome curl_chrome101 'https://www.thenewhumanitarian.org/rss/all.xml'
```

But here is another test using curl-impersonate where it fails:
https://rss-bridge.org/bridge01/?action=display&bridge=FeedMergeBridge&feed_name=thenewhumanitarian&feed_1=https%3A%2F%2Fwww.thenewhumanitarian.org%2Frss%2Fall.xml&feed_2=&feed_3=&feed_4=&feed_5=&feed_6=&feed_7=&feed_8=&feed_9=&feed_10=&limit=&format=Html

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/4b6a8937-474a-407e-a175-a0403f64394c)

So it is a cat/mouse problem.

But if you are so lucky at you have a command that works, you can make a little PHP script `thenewhumanitarian.php` on your Web server somewhere to act as a mini proxy:

```php
<?php
header('Content-Type: application/rss+xml; charset=utf-8');
passthru('curl https://www.thenewhumanitarian.org/rss/all.xml -A "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"');
```

Then call this file from your Web browser to verify that it works, and then if it does, you can subscribe to it in FreshRSS.

**Reply by danburd** [reply]
Thank you. I'm dismayed at how careless some of these website administrators are by including the RSS URL in their bot-detection coverage. For now I have sent them an email explaining the issue, and I subscribed to [a feed of their newsletter](https://us12.campaign-archive.com/feed?u=31c0c755a8105c17c23d89842&id=d842d98289) as a workaround.

I don't think I have a PHP server, but I'll keep that script in mind.

**Reply by Alkarex** [reply]
> I don't think I have a PHP server, but I'll keep that script in mind.

You can put that file in FreshRSS itself, in `./FreshRSS/p/thenewhumanitarian-some-secret.php`

Edit: removed suggestion for Docker (which requires adding `curl` package)
